paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -542,11 +570,18 @@ internal class PaymentSheetViewModel @Inject internal constructor(
                 // Reset after sending event
                 deferredIntentConfirmationType = null
 
-                // Default future payments to the selected payment method. New payment methods won't
-                // be the default because we don't know if the user selected save for future use.
-                when (selection.value) {
-                    is PaymentSelection.New -> null
-                    else -> selection.value
+                /*
+                 * Sets current selection as default payment method in future payment sheet usage. New payment
+                 * methods are only saved if the payment sheet is in setup mode, is in payment intent with setup
+                 * for usage, or the customer has requested the payment method be saved.
+                 */
+                when (val currentSelection = selection.value) {
```

Comment:
@tillh-stripe can you take a look at this?

Reply:
Looks right to me!